### PR TITLE
Add support for metal, vmware, etc. in new CLI #313

### PR DESCRIPTION
**Pull request by toszo**
#313

### CODE REVIEW

File: core/src/epicli/cli/engine/EpiphanyEngine.py [hunk 1]
```diff
@@ -97,13 +98,14 @@ def apply(self):
 
             self.process_infrastructure_docs()
 
-            # Generate terraform templates
-            with TerraformTemplateGenerator(self.cluster_model, self.infrastructure_docs) as template_generator:
-                template_generator.run()
+            if not self.skip_infrastructure:
+                # Generate terraform templates
```
Comment (erzetpe) [inline core/src/epicli/cli/engine/EpiphanyEngine.py]: why to comment this?

File: core/src/epicli/cli/engine/EpiphanyEngine.py [hunk 2]
```diff
@@ -97,13 +98,14 @@ def apply(self):
...
+                with TerraformTemplateGenerator(self.cluster_model, self.infrastructure_docs) as template_generator:
+                    template_generator.run()
 
-            # Run Terraform to create infrastructure
-            with TerraformRunner(self.cluster_model.specification.name) as tf_runner:
-                tf_runner.run()
+                # Run Terraform to create infrastructure
```
Comment (erzetpe) [inline core/src/epicli/cli/engine/EpiphanyEngine.py]: why to comment this?

File: core/src/epicli/cli/epicli.py [hunk 3]
```diff
@@ -54,6 +54,9 @@ def apply_parser(subparsers):
     sub_parser = subparsers.add_parser('apply', description='Applies configuration from file.')
     sub_parser.add_argument('-f', '--file', dest='file', type=str,
                             help='File with infrastructure/configuration definitions to use.')
+    sub_parser.add_argument('-ni', '--no-infra', dest='no_infra', action="store_true",
```
[the remark this answers is not in the document]
Comment (toszo) [reply core/src/epicli/cli/epicli.py]: Sometimes I want to run cli without infra provisioning for aws/azure so it is not just a use case for 'any'

File: core/src/epicli/data/common/validation/core/definitions.yml [hunk 4]
```diff
@@ -19,7 +19,7 @@ provider:
   examples:
   - aws
```
[the remark this answers is not in the document]
Comment (toszo) [reply core/src/epicli/data/common/validation/core/definitions.yml]: Done

File: core/src/epicli/cli/epicli.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (erzetpe) [reply core/src/epicli/cli/epicli.py]: I wonder if using double letter -ni is a good idea. Mainly if you are having single - you should provide one letter or combination of letter, where every single letter has its role.

[the remark this answers is not in the document]
Comment (toszo) [reply core/src/epicli/cli/epicli.py]: It is not apparently - will change it

**Pull request merged**